### ISSUE TITLE
Make libseccomp loading lazy

### ISSUE THREAD

**Issue by j39m** [igo95862/bubblejail#46]
### Your distro name and version

Fedora 37

### Description

This may be a Fedora-specific quirk — feel free to close if not actionable.

I'm experimenting with building `bubblejail` as an RPM package, and it appears that I need to add these to the "build requires" section for COPR build to succeed:

* python3-tomli
* python3-tomli-w (I didn't actually try this one explicitly, just moved it together with tomli)
* python3-pyxdg
* libseccomp

Mainly this seems to be a problem in a hermetic build environment, and only during doc generation, e.g.

```
[3/8] /usr/bin/meson --internal exe --unpickle /builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/meson-private/meson_exe_python3_3497559078f0e25c1cb7a2ffa5851f93cd891ce1.dat
FAILED: docs/bubblejail.1.md 
/usr/bin/meson --internal exe --unpickle /builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/meson-private/meson_exe_python3_3497559078f0e25c1cb7a2ffa5851f93cd891ce1.dat
while executing ['/usr/bin/python3', '-O', '/builddir/build/BUILD/bubblejail-0.7.0/docs/man_generator.py', '--template-dir', '/builddir/build/BUILD/bubblejail-0.7.0/docs/templates', 'cmd']
--- stderr ---
Traceback (most recent call last):
  File "/builddir/build/BUILD/bubblejail-0.7.0/docs/man_generator.py", line 26, in <module>
    from bubblejail.bubblejail_cli import BUBBLEJAIL_CMD
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_cli.py", line 25, in <module>
    from .bubblejail_directories import BubblejailDirectories
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_directories.py", line 28, in <module>
    from .bubblejail_instance import BubblejailInstance, BubblejailProfile
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_instance.py", line 40, in <module>
    from .bubblejail_seccomp import SeccompState
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_seccomp.py", line 42, in <module>
    seccomp_init = import_from_cdll('seccomp_init', (c_uint, ), c_void_p)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_seccomp.py", line 36, in import_from_cdll
    c_function = getattr(libseccomp, func_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/ctypes/__init__.py", line 389, in __getattr__
    func = self.__getitem__(name)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/ctypes/__init__.py", line 394, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: /usr/bin/python3: undefined symbol: seccomp_init
```

**Comment by igo95862** [comment]
Hello,

This issue is caused by the fact that the  `libseccomp` is imported on the module level.

https://github.com/igo95862/bubblejail/blob/09be2c47b94ac58ed33236787f50d9ebf37b10d2/src/bubblejail/bubblejail_seccomp.py#L26

When the man pages are built they pull the metadata from command line parser. Through a chain of import the `bubblejail_seccomp` is loaded and it tries to import the libseccomp.

The obvious solution is to make libseccomp loading lazy. (plus that will speed up execution when libseccomp is not required)

**Comment by j39m** [comment]
Thanks for the speedy response. Just to note, the same applies to the other packages —

to `tomli`:

<details>

```
FAILED: docs/bubblejail.1.md 
/usr/bin/meson --internal exe --unpickle /builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/meson-private/meson_exe_python3_3497559078f0e25c1cb7a2ffa5851f93cd891ce1.dat
while executing ['/usr/bin/python3', '-O', '/builddir/build/BUILD/bubblejail-0.7.0/docs/man_generator.py', '--template-dir', '/builddir/build/BUILD/bubblejail-0.7.0/docs/templates', 'cmd']
--- stderr ---
Traceback (most recent call last):
  File "/builddir/build/BUILD/bubblejail-0.7.0/docs/man_generator.py", line 26, in <module>
    from bubblejail.bubblejail_cli import BUBBLEJAIL_CMD
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_cli.py", line 25, in <module>
    from .bubblejail_directories import BubblejailDirectories
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_directories.py", line 23, in <module>
    from tomli import load as toml_load
ModuleNotFoundError: No module named 'tomli'
```

</details>

and to `pyxdg`:

<details>

```
FAILED: docs/bubblejail.1.md 
/usr/bin/meson --internal exe --unpickle /builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/meson-private/meson_exe_python3_3497559078f0e25c1cb7a2ffa5851f93cd891ce1.dat
while executing ['/usr/bin/python3', '-O', '/builddir/build/BUILD/bubblejail-0.7.0/docs/man_generator.py', '--template-dir', '/builddir/build/BUILD/bubblejail-0.7.0/docs/templates', 'cmd']
--- stderr ---
Traceback (most recent call last):
  File "/builddir/build/BUILD/bubblejail-0.7.0/docs/man_generator.py", line 26, in <module>
    from bubblejail.bubblejail_cli import BUBBLEJAIL_CMD
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_cli.py", line 25, in <module>
    from .bubblejail_directories import BubblejailDirectories
  File "/builddir/build/BUILD/bubblejail-0.7.0/redhat-linux-build/src/bubblejail/bubblejail_directories.py", line 25, in <module>
    from xdg import IniFile
ModuleNotFoundError: No module named 'xdg'
```

</details>

**Comment by igo95862** [comment]
Are you sure you have `tomli` and `pyxdg` installed? Otherwise there should not be any import problems. Try running the python interpreter and importing them manually.

**Comment by j39m** [comment]
I can't speak for the exact state of the COPR build infra, but I suspect it starts with an extremely barebones installation that does _not_ have those packages installed. I can only say that when I add `tomli` and `pyxdg` to the "build requires" section, the build succeeds.

**Comment by igo95862** [comment]
I see.

Because 0.7.0 man page is built from the argument parser meta it will import the command line module which will pull imports for all other modules like `tomli` and `pyxdg`.

The solution is to separate the argument parser metadata to separate module.

**Comment by j39m** [comment]
This is  fixed by bfe2730, yes?

**Comment by igo95862** [comment]
> This is fixed by [bfe2730](https://github.com/igo95862/bubblejail/commit/bfe2730671804690b61fd214ce2521b8ec6bd1f3), yes?

The build requirements were fixed by separating the command line arguments metadata in to separate package. However, I still plan on making the libseccomp loading lazy as it slows down execution of some commands were it is not needed like `list`.

**Comment by igo95862** [comment]
Implemented with b83344c5e5bf569dd2c051e5cee1e7e36dc5bfec
